datasources-agroal/src/main/java/org/wildfly/extension/datasources/agroal/DataSourceService.java

```diff
@@ -27,6 +27,7 @@
 import io.agroal.api.transaction.TransactionIntegration;
 import io.agroal.narayana.NarayanaTransactionIntegration;
 import org.ietf.jgss.GSSException;
+import org.jboss.as.server.deployment.DelegatingSupplier;
```

Comment:
I'd prefer not to use the server module as a utility lib.  

Extensions can use the deployment package in DUPs, but this isn't a DUP. (At some point we should factor out that package into its own module.)

AIUI using this just eliminates the need for some null checks Optional.isPresent checks.

Reply:
Fixed @bstansberry  . There was introduced a new DelegatingSupplier class in EE subsystem and AGROAL subsystem was rewritten to don't use DelegatingSupplier at all.
We're using DelegatingSupplier not because of Optional.isPresent() checks but because there are some places where supplier injection happens later than at the service installation time.